presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -863,7 +884,12 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, PreferredProperties p
                     }
                     else {
                         source = withDerivedProperties(
-                                partitionedExchange(idAllocator.getNextId(), REMOTE_STREAMING, source.getNode(), createPartitioning(sourceSymbols), Optional.empty()),
+                                partitionedExchange(
```

Comment:
per offline discussion, today we don't get any benefits by materialize the exchange for SEMI JOIN. So maybe we don't need to do it for now? -- We can do it later when we support stage level retry.